### PR TITLE
Feature/compare two countries

### PR DESCRIPTION
**Pull request by madisonlemmond**
This feature allows users to compare the COVID-19 data of two countries. The data is displayed in charts.

### CODE REVIEW

**Comment by madisonlemmond** [comment]
To test the comparison feature, go to localhost:4200/comparison/:country/:countryTwo

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,15 +1,31 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 
+import { AngularFireModule } from '@angular/fire';
+import { AngularFirestoreModule } from '@angular/fire/firestore';
+import { AngularFireStorageModule } from '@angular/fire/storage';
+import { AngularFireAuthModule } from '@angular/fire/auth';
+import { environment } from '../environments/environment';
+
```
Comment (wSedlacek) [inline src/app/app.module.ts]: AngularFire should not be included as part of this PR

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
 import { WorldModule } from './features/world/world.module';
 
 @NgModule({
   declarations: [AppComponent],
-  imports: [BrowserModule, SharedModule, AppRoutingModule, CountryModule, WorldModule],
+  imports: [
+    AngularFireModule.initializeApp(environment.firebase),
+    AngularFirestoreModule, // firestore
+    AngularFireAuthModule, // auth
+    AngularFireStorageModule,
```
Comment (wSedlacek) [inline src/app/app.module.ts]: AngularFire should not be included as part of this PR

File: src/app/features/country/components/country-comparison-page/country-comparison-page.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<ng-container *ngIf="countryTwo$ | async; let countryTwo">
+  <app-country-comparison
+    [country]="country$"
+    (countryChanged)="switchCountry($event, countryTwo)"
+    [countryTwo]="countryTwo$"
+    (countryTwoChanged)="switchCountryTwo($event)"
+  ></app-country-comparison>
+</ng-container>
```
Comment (wSedlacek) [inline src/app/features/country/components/country-comparison-page/country-comparison-page.component.html]: It seems unnecessary to make a `app-country-comparison` and a `app-country-comparison-page`.
`app-country-comparison` is only used in one place so it should be inlined here

File: src/app/features/country/components/country-comparison-page/country-comparison-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  styleUrls: ['./country-comparison-page.component.scss']
+})
+export class CountryComparisonPageComponent {
+  constructor(private readonly route: ActivatedRoute, private readonly router: Router) {}
+  public readonly country$ = this.route.params.pipe(map(({ country }) => country as string));
+  public readonly countryTwo$ = this.route.params.pipe(map(({ countryTwo }) => countryTwo as string));
+
+  public switchCountry(country: string, countryTwo: string): void {
+    this.router.navigate([`comparison/${country}/${countryTwo}`], { relativeTo: this.route });
+  }
```
Comment (wSedlacek) [inline src/app/features/country/components/country-comparison-page/country-comparison-page.component.ts]: Not a fan of the absolute pathing here. I think a relative path would make it easier to change the path later.

File: src/app/features/country/country-page.module.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { DataComponent } from './components/data/data.component';
+import { CountryPageComponent } from './country-page.component';
+import { CountryComparisonComponent } from './components/country-comparison/country-comparison.component';
+import { CountryComparisonPageComponent } from './components/country-comparison-page/country-comparison-page.component';
+
+
+@NgModule({
+  imports: [SharedModule, MatSelectModule, NgxChartsModule],
+  declarations: [CountryComponent, CountryPageComponent, CountryComparisonComponent, CountryComparisonPageComponent, MenuComponent, DataComponent],
+  exports: [MenuComponent, DataComponent, CountryPageComponent, CountryComparisonComponent, CountryComparisonPageComponent],
```
Comment (wSedlacek) [inline src/app/features/country/country-page.module.ts]: No exports are needed here unless these components are used in other modules.

**Review by wSedlacek — COMMENTED** [review]
Overall I think you did a great job.
I have made some changes to clean up things a bit.
Take a look over the commit and if you are happy with how things are then go ahead and merge.

**Review by wSedlacek — APPROVED** [review]

**Pull request merged**